src/controllers/webviewController.ts

```diff
@@ -61,6 +62,10 @@ export class WebviewPanelController implements vscode.Disposable {
                     // angular component to show up
                     setTimeout(async () => await this.queryRunner.refreshQueryTab(), 1700);
                 }
+            // if focus is on text editor
+            } else if (p.webviewPanel.visible && !p.webviewPanel.active) {
+                this._isPanelFocused = true;
```

Comment:
this code block seems weird to me.  The comment says "focus is on the text editor" the condition checks that the panel is not active, then we set panel is focused to true?

maybe I'm missing something but the three inconsistent.  Could you please explain this code block a little?  Thanks!

Reply:
There are 3 cases in this scenario -
1. The webview panel is visible and active. 
2. The webview panel is not visible and not active. (That means the results tab was changed)
2. The webview panel is visible but not active. (That means the editor has focus). 

I set the `isFocused` variable in the second and third cases because this event is propagated before the actual `active` and `focus` properties are set. The way the second case can happen is also 2 ways -
1. it just got focus from the text editor (which is case 2)
2. it just got focus from a tab change

I keep track of the first part by using the `isPanelFocused` variable to detect what kind of change it was. Perhaps I can name it to `didTabChange` for better readability.